paddle/fluid/operators/distributed/grpc_server.cc

```diff
@@ -156,6 +156,98 @@ class RequestGet final : public RequestBase {
   ServerAsyncResponseWriter<::grpc::ByteBuffer> responder_;
 };
 
+class RequestGetMonomerVariable final : public RequestBase {
```

Comment:
I think maybe no need to add a new request since you are implementing a server, which can control the barrier.

Reply:
The older logic should be changed to use VariableBarrier.